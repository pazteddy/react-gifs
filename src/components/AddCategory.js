import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e)=>{
    // console.log(e.target.value);
    setInputValue(e.target.value);
  }
  const onHandleSubmit = (e) =>{
    // Evita que se recargue la pagina
      e.preventDefault();
      // console.log('onHandleSubmit ',inputValue);
      if(inputValue.trim().length > 2){
        setCategorias(cats => [inputValue,...cats]);
        setInputValue('');
      }
      
  };
  return (
    <form onSubmit={ onHandleSubmit }>
      <input 
        type="text" 
        name="" 
        id=""
        value={inputValue}
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired
}
