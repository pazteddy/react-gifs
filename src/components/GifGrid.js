import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
 import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {categoria}) => {
  // const [images, setImages] = useState([]);
  // //Solo se llame una vez para traer los gifs asi se actulice el componente
  // useEffect(() => {
  //   getGifs(categoria)
  //   .then( imgs => setImages( imgs));
  // }, [ categoria ]);

  //getGifs();

  const { data:images,loading } = useFetchGifs(categoria);
  
  return (
    <>
    <h3 className="animate__animated animate__fadeIn">{categoria}</h3>
    { loading && <p className="animate__animated animate__flash">Loading</p> }
    <div className="card-grid">
        {
          images.map( img => (
            <GifGridItem 
            key = {img.id}
            {...img}/>
          ))
        }
    </div>
    </>
  )
}
