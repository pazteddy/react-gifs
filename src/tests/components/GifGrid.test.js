import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe(" Pruebas de <GifGrid/>", () =>{
  const categoria = 'Heroe Academia';
  test('Debe mostrar correctamente <GifGrid/>', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading : true
    });
    const wrapper = shallow(<GifGrid categoria={categoria}/>);
    expect( wrapper ).toMatchSnapshot();
  });
  test('Deben mostrar items cuando se cargan imÃ¡genes useFecthGifs', () => {
    const gifs=[{
      id:'ABC',
      url: 'https//localhost/cosa.gif',
      title:'Cualquier cosa'
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading : false
    });
    const wrapper = shallow(<GifGrid categoria={categoria}/>);
    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('p').exists()).toBe(false);
    expect( wrapper.find('GifGridItem').length).toBe( gifs.length );
  });
  
});