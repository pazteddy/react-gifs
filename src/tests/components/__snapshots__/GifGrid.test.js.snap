// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas de <GifGrid/> Debe mostrar correctamente <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Heroe Academia
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[` Pruebas de <GifGrid/> Deben mostrar items cuando se cargan im√°genes useFecthGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Heroe Academia
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https//localhost/cosa.gif"
    />
  </div>
</Fragment>
`;
